HW1/P5/P5.txt

```diff
@@ -0,0 +1,156 @@
+ [u'Harvard_University', u'Tommy_Lee_Jones', u'2005_Cannes_Film_Festival', u'Kevin_Bacon'],
+ [u'Harvard_University', u'Pine_Manor_College', u'Dennis_Lehane', u'Kevin_Bacon'],
+ [u'Harvard_University', u'Natalie_Portman', u'Erd\u0151s\u2013Bacon_number', u'Kevin_Bacon'],
+ [u'Harvard_University', u'Natalie_Portman', u"Paris,_je_t'aime", u'Kevin_Bacon'],
+ [u'Harvard_University', u'Hasty_Pudding_Woman_of_the_Year', u'1958', u'Kevin_Bacon'],
+ [u'Harvard_University', u'List_of_U.S._colleges_and_universities_by_endowment', u'July_8', u'Kevin_Bacon'],
+ [u'Harvard_University', u'Jack_Lemmon', u'Gene_Hackman', u'Kevin_Bacon'],
+ [u'Harvard_University', u'2007', u'1958', u'Kevin_Bacon'],
+ [u'Harvard_University', u'University_of_Virginia', u'Tiki_Barber', u'Kevin_Bacon'],
+ [u'Harvard_University', u'Tommy_Lee_Jones', u'Dustin_Hoffman', u'Kevin_Bacon'],
+ [u'Harvard_University', u'Richard_Nixon', u'Apollo_13', u'Kevin_Bacon'],
+ [u'Harvard_University', u'The_Firm_(1993_film)', u'Gene_Hackman', u'Kevin_Bacon'],
+ [u'Harvard_University', u'2006', u'July_8', u'Kevin_Bacon'],
+ [u'Harvard_University', u'University_of_Oregon', u"National_Lampoon's_Animal_House", u'Kevin_Bacon'],
+ [u'Harvard_University', u'August_30', u'1958', u'Kevin_Bacon'],
+ [u'Harvard_University', u'Good_Will_Hunting', u"Paris,_je_t'aime", u'Kevin_Bacon'],
+ [u'Harvard_University', u'Matt_Damon', u'Gene_Hackman', u'Kevin_Bacon'],
+ [u'Harvard_University', u'September_28', u'1958', u'Kevin_Bacon'],
+ [u'Harvard_University', u'Boston_Brahmin', u'Endicott_Peabody_(educator)', u'Kevin_Bacon'],
+ [u'Harvard_University', u'September_2', u'July_8', u'Kevin_Bacon'],
+ [u'Harvard_University', u'The_Da_Vinci_Code', u'Thriller_(genre)', u'Kevin_Bacon'],
+ [u'Harvard_University', u'February_26', u'July_8', u'Kevin_Bacon'],
+ [u'Harvard_University', u'Jack_Lemmon', u'Mister_Roberts_(film)', u'Kevin_Bacon'],
+ [u'Harvard_University', u'Texas_A&M_University', u'A_Few_Good_Men', u'Kevin_Bacon'],
+ [u'Harvard_University', u'George_W._Bush', u'July_8', u'Kevin_Bacon'],
+ [u'Harvard_University', u'Jack_Lemmon', u'Dustin_Hoffman', u'Kevin_Bacon'],
+ [u'Harvard_University', u'February_19', u'1958', u'Kevin_Bacon'],
+ [u'Harvard_University', u'London', u'Elvis_Costello', u'Kevin_Bacon'],
+ [u'Harvard_University', u'2003', u'July_8', u'Kevin_Bacon'],
+ [u'Harvard_University', u'2006', u'1958', u'Kevin_Bacon'],
+ [u'Harvard_University', u'May_21', u'1958', u'Kevin_Bacon'],
+ [u'Harvard_University', u'John_Kerry', u'John_Edwards_presidential_campaign,_2008', u'Kevin_Bacon'],
+ [u'Harvard_University', u'2007', u'2007_in_film', u'Kevin_Bacon'],
+ [u'Harvard_University', u'Boston,_Massachusetts', u'South_Boston,_Boston,_Massachusetts', u'Kevin_Bacon'],
+ [u'Harvard_University', u'Tommy_Lee_Jones', u'Screen_Actors_Guild_Award_for_Outstanding_Performance_by_a_Cast_in_a_Motion_Picture', u'Kevin_Bacon'],
+ [u'Harvard_University', u'September_16', u'July_8', u'Kevin_Bacon']]
```

Comment:
I didn't get to the extra credit question, because my algo was not optimized enough. It's nice to see how you tackled the extra extra credit, still absorbing it...
Also, I appreciate you listed all the paths.